synapse/storage/databases/main/user_directory.py

```diff
@@ -228,10 +228,7 @@ def _get_next_batch(
             is_in_room = await self.is_host_joined(room_id, self.server_name)
 
             if is_in_room:
-                is_public = await self.is_room_world_readable_or_publicly_joinable(
-                    room_id
-                )
-
+                # TODO this leaks per-room profiles!
```

Comment:
this seems to be in conflict with the newsfile :thinking: 

I think I don't really understand what the TODO means. How does it leak them?

Perhaps it's ok if you're going to be updating this soon, but otherwise a few more words in this comment might be helpful.

Reply:
I'll expand the comment. For completeness here:

`get_users_in_room_with_profiles` reads from the `room_memberships` table. That contains a `display_name` and `avatar_url` specific to this room (it doesn't read from the `profiles` table). With my change, we no longer upsert `user_directory` and `user_directory_search` rows for local users in this function, but we *do* continue to do so for remote users.

I meant the TODO as a marker to track where the data we don't want to leak comes from.